src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java

```diff
@@ -151,33 +151,34 @@ public ActivityServiceImpl(ActivityRepository activityRepository, RecordsService
     public Page<ProviderRecordDTO> getPartnerActivitiesForCurrentUser(Pageable pageable) {
         UserProfile userProfile = userService.getCurrentUserProfile();
         Set<UserGroup> userGroups = userProfile.getUserGroups();
-        Page<Organization> organizations;
         if (userGroups == null || userGroups.isEmpty()) {
-            organizations = providerRecordsRepository
+            return providerRecordsRepository
                 .findAllWithFilters(userProfile, null, new ProviderFilterDTO(), null, pageable);
         } else {
-            organizations = organizationService.findAllByUserGroups(pageable, new ArrayList<>(userGroups));
+            return filterProviderRecords(
+                organizationService.findAllByUserGroups(pageable, new ArrayList<>(userGroups))
+                .map(this::getProviderRecordDTO)
```

Comment:
filter is not needed here, because getProviderRecordDTO is removing the excluded fields